### PR TITLE
Update Maven settings per Sonatype Release (3.x branch)

### PR DESCRIPTION
**Pull request by tdonohue**
Because XOAI 3.x hasn't been released in some time, the Maven POMs are outdated per recent Sonatype requirements. This POM makes those updates.

Based on similar PRs to the DSpace lang packs:
* https://github.com/DSpace/dspace-xmlui-lang/pull/27
* https://github.com/DSpace/dspace-api-lang/pull/19

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,100 +4,202 @@
...
     <name>XOAI : OAI-PMH Java Toolkit</name>
     <version>3.2.11-SNAPSHOT</version>
+    <url>https://github.com/dspace/xoai</url>
 
-    <parent>
-        <groupId>org.sonatype.oss</groupId>
-        <artifactId>oss-parent</artifactId>
-        <version>9</version>
-    </parent>
+    <prerequisites>
```
Comment (mwoodiupui) [inline pom.xml]: We should use the enforcer plugin for this.  I'm told that <prerequisites> is for Maven plugin projects only.

File: pom.xml [hunk 2]
```diff
@@ -4,100 +4,202 @@
...
-                    </execution>
-                </executions>
-            </plugin>
         </plugins>
     </build>
 
+    <profiles>
+       <!--
+         The 'release' profile is used by the 'maven-release-plugin' (see above)
+         to actually perform a DSpace lang packs release to Maven central.
```
Comment (mwoodiupui) [inline pom.xml]: Uh, "DSpace lang packs"?  I thought we were building XOAI.

File: pom.xml (hunk 1 above)
Comment (tdonohue) [reply pom.xml]: Good point, I forgot about this. Fixed in latest commit

File: pom.xml (hunk 2 above)
Comment (tdonohue) [reply pom.xml]: I overlooked that copy & paste error. Fixed in latest commit

File: pom.xml (hunk 1 above)
Comment (tdonohue) [reply pom.xml]: Good point, I forgot about this. Fixed in latest commit

File: pom.xml [hunk 3]
```diff
@@ -36,6 +32,9 @@
                                     <requireJavaVersion>
                                         <version>${java.version}</version>
                                     </requireJavaVersion>
+                                    <requireMavenVersion>
+                                        <version>[3.0.5,)</version>
```
Comment (paulo-graca) [inline pom.xml]: is this version correct?

Comment (tdonohue) [reply pom.xml]: These settings were copied from our DSpace/DSpace POM: https://github.com/DSpace/DSpace/blob/master/pom.xml#L71-L73

So, yes, they should work fine. Most people should have a much newer version of Maven than 3.0.5 though at this point.

**Comment by tdonohue** [comment]
Tested this again today and a `mvn release:prepare -DdryRun=true` succeeds now. So, at least the dryRun is working. I'll only be able to test the full release once this is merged.  Merging now as this has had reviews & will allow me to try a new release of XOAI 3.x

**Comment by tdonohue** [comment]
After testing a full Maven release, I found this POM was missing a required `<description>` field per the [Sonatype POM requirements](https://central.sonatype.org/pages/requirements.html).

Therefore, one additional minor commit was pushed to the `3.x` branch directly: https://github.com/DSpace/xoai/commit/dd97db3704588d88f452d87ad38a476a0f2c5d1a

**Pull request merged**